File: backend/DoctorLicense.Domain/Entities/Doctor.cs
namespace DoctorLicense.Domain.Entities;


public class Doctor
{
    public int Id { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Specialization { get; set; } = string.Empty;
    public string LicenseNumber { get; set; } = string.Empty;
    public DateTime LicenseExpiryDate { get; set; }
    public string Status { get; set; } = "Active";
    public DateTime CreatedDate { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime? DeletedDate { get; set; }
}

File: frontend/types/doctor.ts

export interface Doctor {
  id: number;
  fullName: string;
  email: string;
  specialization: string;
  licenseNumber: string;
  licenseExpiryDate: string;
  status: 'Active' | 'Expired' | 'Suspended' | 'Pending';
  createdDate: string;
}

export interface DoctorListResponse {
  doctors: Doctor[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

export interface CreateDoctorDto {
  fullName: string;
  email: string;
  specialization: string;
  licenseNumber: string;
  licenseExpiryDate: string;
}

export interface UpdateDoctorDto extends CreateDoctorDto {
  id: number;
}